transformation/transformationPlugins/jsonMessageTransformers/jsonTypeMappingsSanitizationTransformerProvider/src/test/java/org/opensearch/migrations/replay/TypeMappingsSanitizationProviderTest.java

```diff
@@ -79,4 +87,79 @@ public void testSimpleTransform() throws JsonProcessingException {
                 JsonNormalizer.fromObject(resultFromNullConfig));
         }
     }
+
+    @Test
+    public void testMappingWithoutTypesAndLatestSourceInfoDoesNothing() throws Exception {
+        var testString = TestRequestBuilder.makePutIndexRequest("commingled_docs", true, false);
+        var fullTransformerConfig =
+            Map.of("sourceProperties",
+                Map.of("version",
+                    Map.of("major",  (Object) 6,
+                        "minor", (Object) 10)));
+        var transformer = new TypeMappingSanitizationTransformerProvider().createTransformer(fullTransformerConfig);
+        var resultObj = transformer.transformJson(OBJECT_MAPPER.readValue(testString, LinkedHashMap.class));
+        Assertions.assertEquals(JsonNormalizer.fromString(testString), JsonNormalizer.fromObject(resultObj));
+    }
+
+    @Test
+    public void testTypeMappingsWithSourcePropertiesWorks() throws Exception {
```

Comment:
I'm struggling to understand the difference between this and the `testMappingWithoutTypesAndLatestSourceInfoDoesNothing` test in terms of what they're verifying.  I see that we're passing the regex in here and the version is different, but the result is the same either way?  Maybe I'm missing something here...